codalab/lib/bundle_store.py

```diff
@@ -80,6 +60,336 @@ def reset(self):
+
+    def add_partition(self, target, new_partition_name):
+        """
+        BalancedMultiDiskBundleStore specific method. Add a new partition to the bundle store. The "target" is actually a symlink to
+        the target directory, which the user has configured as the mountpoint for some desired partition.
+        """
+        target = os.path.abspath(target)
+        new_partition_location = os.path.join(self.partitions, new_partition_name)
+
+        print >> sys.stderr, "Adding new partition as %s..." % new_partition_location
+        path_util.soft_link(target, new_partition_location)
+
+        mdata = os.path.join(new_partition_location, BalancedMultiDiskBundleStore.DATA_SUBDIRECTORY)
+
+        try:
+            path_util.make_directory(mdata)
+        except Exception, e:
+            print >> sys.stderr, e
+            print >> sys.stderr, "Could not make directory %s on partition %s, aborting" % (mdata, target)
+            sys.exit(1)
+
+        self.nodes.append(new_partition_name)
+        new_mtemp = os.path.join(new_partition_location, BalancedMultiDiskBundleStore.TEMP_SUBDIRECTORY)
+        path_util.make_directory(new_mtemp)
+
+        print >> sys.stderr, "Successfully added partition '%s' to the pool." % new_partition_name
+
+    def reset(self):
+        """
+        Delete all stored bundles and then recreate the root directories.
+        """
+
+        pass
+
+        '''
+        # Do not run this function in production!
+        path_util.remove(self.partitions)
+        self.initialize_store()
+        '''
+
+    def __get_num_partitions(self):
+        """
+        Returns the current number of disks being used by this BalancedMultiDiskBundleStore.
+        This is calculated as the number of directories in self.partitions
+        """
+        return reduce(lambda dirs, _: len(dirs), path_util.ls(self.partitions))
+
+
+    @require_partitions
+    def rm_partition(self, partition):
+        """
+        Deletes the given partition entry from the bundle store, and purges the lru cache. Does not move any bundles
+        """
+
+        if self.__get_num_partitions() == 1:
+            """
+            Prevent foot-shooting
+            """
+            print >> sys.stderr, "Error, cannot remove last partition. If you really wish to delete CodaLab, please run the following command:"
+            print >> sys.stderr, "      rm -rf %s" % self.codalab_home
+            return
+
+        partition_abs_path = os.path.join(self.partitions, partition)
+
+        try:
+            print partition_abs_path
+            path_util.check_isvalid(partition_abs_path, 'rm-partition')
+        except:
+            print >> sys.stderr, "Partition with name '%s' does not exist. Run `cl ls-partitions` to see a list of mounted partitions." % partition
+            sys.exit(1)
+
+        print >> sys.stderr, "Unlinking partition %s from CodaLab deployment..." % partition
```

Comment:
I think that would be unnecessary. Since it doesn't do anything destructive to bundles, if someone removes a partition by mistake, all he/she has to do is to re-add it.